flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTable.java

```diff
@@ -356,7 +373,7 @@ public boolean markSlotActive(AllocationID allocationId) throws SlotNotFoundExce
 	 * @return True if the slot could be marked inactive
 	 */
 	public boolean markSlotInactive(AllocationID allocationId, Time slotTimeout) throws SlotNotFoundException {
-		checkInit();
+		checkStartedOrClosing();
```

Comment:
Why do we allow the closing state here? I think we cannot do this because this might introduce a potential race condition between the `timerService` being stopped and registering a new timeout on it.

Reply:
As discussed offline, enabling this check would need at least disabling message handling after `RpcEndpoint#onStop` to not accept disconnect messages from job master after `taskExecutor#onStop`. On the other hand, requiring only start method call (not `CREATED`) should be enough because calling it in non-running state will register a timeout in worse case or observe no slot. The issue with concurrent access in `timerService#stop` should be resolved anyways.